sub-packages/bionemo-esm2/tests/bionemo/esm2/model/test_stop_and_go.py

```diff
@@ -108,3 +113,73 @@ def setup_model(cls, mode: Mode) -> tuple[pl.LightningModule, pl.LightningDataMo
+        callbacks[Mode.STOP][testing_callbacks.SignalAfterGivenStepCallback] = (
+            testing_callbacks.SignalAfterGivenStepCallback(stop_step=2, signal_=signal.SIGUSR2)
+        )
+
+        return callbacks
+
+    @override
+    @classmethod
+    def stop(cls) -> None:
+        # The PreemptionCallback exits the process with sys.exit(0) after the checkpoint is saved. We obviously don't
+        # want that here, so we catch the SystemExit exception and make sure it was called appropriately.
+        with pytest.raises(SystemExit) as pytest_wrapped_e:
+            super().stop()
+
+        assert pytest_wrapped_e.type is SystemExit
+        assert pytest_wrapped_e.value.code == 0
+
+    @pytest.mark.parametrize(
+        "callback_type",
+        [
+            testing_callbacks.LearningRateCallback,
+            testing_callbacks.GlobalStepStateCallback,
+            testing_callbacks.ConsumedSamplesCallback,
+            testing_callbacks.OptimizerStateCallback,
+            testing_callbacks.TrainInputCallback,
+            testing_callbacks.TrainOutputCallback,
+            testing_callbacks.TrainLossCallback,
+            testing_callbacks.ValidInputCallback,
+            testing_callbacks.ValidOutputCallback,
+            testing_callbacks.ValidLossCallback,
+        ],
+    )
+    def test_stop_and_go_consistency(self, callback_type):
+        if callback_type in [
+            testing_callbacks.ValidInputCallback,
+            testing_callbacks.ValidLossCallback,
+            testing_callbacks.ValidOutputCallback,
+        ]:
+            # On resumption from a checkpoint that wasn't created at the end of validation, the validation interval is
+            # shifted in the subsequent training jobs. See this slack thread for more details:
+            # https://nvidia.slack.com/archives/C074Z808N05/p1733171223813409
+            pytest.xfail(reason="Currently seeing issues in validation timing with PreemptionCallback")
+        super().test_stop_and_go_consistency(callback_type)
+
+    @pytest.mark.skip(reason="We don't expect the STOP variant to hit on_valid_epoch_end before stopping.")
+    def test_train_val_init_consumed_samples(self):
+        pass
+
+    def test_all_valid_batch_inputs_are_identical(self):
+        """A watered-down version of test_stop_and_go_consistency's ValidInputCallback that only checks whether the
+        first batches are the same, not the over length."""
+
+        valid_inputs_interrupted = stop_and_go.get_callback(
+            self.callbacks, Mode.RESUME, testing_callbacks.ValidInputCallback
+        ).data
+        valid_inputs_continuous = stop_and_go.get_callback(
+            self.callbacks, Mode.CONTINUOUS, testing_callbacks.ValidInputCallback
+        ).data
+
+        min_len = min(len(valid_inputs_interrupted), len(valid_inputs_continuous))
+        assert min_len
+        recursive_assert_approx_equal(valid_inputs_interrupted[:min_len], valid_inputs_continuous[:min_len])
```

Comment:
Can you remind me of which validation is missing in the stop and resume route? This will help me understand the logic here.